src/nfa/nfa.cc

```diff
@@ -555,3 +555,36 @@ bool Nfa::is_identical(const Nfa& aut) const {
     }
     return delta == aut.delta;
 }
+
+Nfa& Nfa::unite_nondet_with(const mata::nfa::Nfa& aut) {
+    const size_t num_of_states{ this->num_of_states() };
+    const size_t aut_num_of_states{ aut.num_of_states() };
+    const size_t new_num_of_states{ num_of_states + aut_num_of_states };
+    auto renumber_states = [&](State st) {
+        return st + num_of_states;
+    };
```

Comment:
I would move this after the three following `if` checks.

Reply:
Done.

Reply:
Did you forget to push your changes? Because I think all the resolved stuff is not pushed :D

Reply:
Yes. The changes are waiting locally on my machine. I was waiting on how we resolve the remaining issues to see whether something else needs to change, too.